src/app/components/mnoe-api/provisioning.svc.coffee

```diff
@@ -82,32 +87,37 @@ angular.module 'mnoEnterpriseAngular'
       else
         _self.updateSubscription(subscription, currency)
 
-    @fetchSubscription = (id) ->
+    @fetchSubscription = (id, cart) ->
       deferred = $q.defer()
       MnoeOrganizations.get().then(
         (response) ->
-          MnoeApiSvc.one('/organizations', response.organization.id).one('subscriptions', id).get().then(
+          params = if cart then { 'subscription[cart_entry]': 'true' } else {}
+          MnoeApiSvc.one('/organizations', response.organization.id).one('subscriptions', id).get(params).then(
             (response) ->
               deferred.resolve(response)
           )
       )
       return deferred.promise
 
-    @getSubscriptions = () ->
+    @getSubscriptions = (params = {}, cart = false) ->
+      return _self.subscriptionsPromise if cart && _self.subscriptionsPromise?
+
       deferred = $q.defer()
       MnoeOrganizations.get().then(
         (response) ->
-          subscriptionsApi(response.organization.id).getList().then(
+          subscriptionsApi(response.organization.id).getList(params).then(
             (response) ->
               deferred.resolve(response)
           )
       )
+      _self.subscriptionsPromise = deferred.promise if cart
```

Comment:
Since this is only used in the cart, maybe we should rename it `cartSubscriptionsPromise`